### ISSUE TITLE
Question about a PR to support streaming from a GLSurfaceView

### ISSUE THREAD

**Issue by pascalbaljet** [pedroSG94/rtmp-rtsp-stream-client-java#210]
Hi Pedro,

First of all, thanks again for this great library! I've been using it for many months now and it just gets better and better. I was wondering if you'd be interested in working together on a PR to stream from a GLSurfaceView. In one of the issues I found this commit which seems exactly what I need: https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/a3ff2ca07cf0952b1d68ab655f1d04b08804a90d

I've checked out that commit, ran it on a Samsung S7 and tried to stream it to a Wowza RTMP endpoint. I did not receive any video content but couldn't find any error in the logs either (streaming from the camera works perfectly though). I thought it might be helpful to implement this as a feature, it seems there are more people needing this:

https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/issues/160
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/issues/136
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/issues/117
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/issues/73

Would you like me to submit a PR for this, copying most stuff from the commit (https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/a3ff2ca07cf0952b1d68ab655f1d04b08804a90d) and update it to the current state of the library?

**Comment by pedroSG94** [comment]
Yes. That mode was discarded because I can't find a way to do it usable for all without a hard implementation. Using an abstract class that you only need extend or similar.
Also we need find a way to send a constant frame rate because by default GLSurfaceView only draw when you have something to refresh and that produce stream stop on some servers like in Facebook or YouTube that need a constant frame rate without stop (same case if you record a video).

Of course if you can find an usable way (because my commit was a demo more than an usable mode)  you are totally welcome to PR and I will merge it after test :D

**Comment by pascalbaljet** [comment]
I've copied the demo from https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/a3ff2ca07cf0952b1d68ab655f1d04b08804a90d and added an example from [Google's Grafika repo](https://github.com/google/grafika/blob/master/app/src/main/java/com/android/grafika/RecordFBOActivity.java) which demonstrates *efficient display + recording of OpenGL rendering using an FBO*.

I've added the activity as [FrameBufferActivity](https://github.com/pascalbaljet/rtmp-rtsp-stream-client-java/blob/glsurfaceview-support/app/src/main/java/com/pedro/rtpstreamer/glsurfaceviewexample/FrameBufferActivity.java). I think the magic is more or less in this activity and in [TextureMovieEncoder2](https://github.com/pascalbaljet/rtmp-rtsp-stream-client-java/blob/glsurfaceview-support/app/src/main/java/com/pedro/rtpstreamer/glsurfaceviewexample/TextureMovieEncoder2.java) and [VideoEncoderCore](https://github.com/pascalbaljet/rtmp-rtsp-stream-client-java/blob/glsurfaceview-support/app/src/main/java/com/pedro/rtpstreamer/glsurfaceviewexample/VideoEncoderCore.java).  Is there any possibility to get this to work with your `GetH264Data` interface?

**Comment by pascalbaljet** [comment]
Hmm maybe it's easier to modify the `openglexample`. I'm gonna try to strip the `Camera1ApiManager` stuff out of `Camera1Base` and only make use of `OpenGlView` and `ManagerRender`.

**Comment by pedroSG94** [comment]
You are still using FBO with opengl if you use managerrender:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/encoder/src/main/java/com/pedro/encoder/input/gl/render/ManagerRender.java#L38

I can do you a glsurfaceview demo that change color every second to test if you want. The problem is how do to work it with a custom glsurfaceview developed by other with minimal changes.

**Comment by pascalbaljet** [comment]
It would be great if you could do the GLSurfaceView demo, then I'll do my best to make it as versatile as possible!

**Comment by pascalbaljet** [comment]
@pedroSG94 I managed to get what I need with the `AndroidViewFilterRender`. Is there any way to get rid of the camera frames underneath the filter and only stream the content of the `AndroidViewFilterRender`? That would be awesome! Strictly speaking this works but it's a waste of resources to leave the camera open and fetch the data constantly.

**Comment by pedroSG94** [comment]
Yes, you can do it. But I think that use display mode is better for that way. Also remember that androidviewfilterrender is limited because some view like glsurfaceview can't be streamed (because in that view you can't take canvas).
Sorry for take so many days to do that demo but now I'm on holydays and I haven't time to do it. I will try do it when my holydays finish or when I have time.

**Comment by pedroSG94** [comment]
I did your demo that stream doing a change color every second:
[test.zip](https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/files/2323636/test.zip)

**Comment by pascalbaljet** [comment]
Thank you! I've got a working solution and will try to incorporate your demo, hopefully I can submit a PR soon.
